src/index.ts

```diff
@@ -596,7 +596,7 @@ class Envoy {
         fn_name: call_spec.function,
         payload: decodedArgs,
         cap: null, // Note: when null, this call will pass when the agent has an 'Unrestricted' status (this includes all calls to an agent's own chain)
-        provenance: Buffer.from(call_spec.cell_id[1]),
+        provenance: anonymous ? Codec.AgentId.decodeToHoloHash(agent_id) : Buffer.from(call_spec.cell_id[1]), // when anonymous, the agent pubkey will not be the cell's agent pubkey
```

Comment:
How can an anonymous zome call go through properly if there is a mismatched provenance passed and no cap token? Did you forget to include a cap token here?

Reply:
No, this approach is designed to not require cap tokens to be passed in for anonymous users.  The DNA creates an unrestricted cap grant for the allowed (read-only) functions, then if the zomeCall has unrestricted access any user when invoked (ie: whenever the provenance is not the agent of the cell / the anonymous case), it successfully completes the zomeCall.